### ISSUE TITLE
Everyone/Anyone please test new Sandstorm Wekan features

### ISSUE THREAD

**Issue by xet7** [wekan/wekan#2208]
Hi everyone,
please try todays huge new Sandstorm Wekan features, preferably with new Sandstorm dev install, and if you have existing data **please make backup first**, and test your restore works.

I have made visible some Standalone Wekan features that were hidden in Sandstorm Wekan. This make Sandstorm Wekan much more close to Standalone Wekan in features. I have not removed anything from Sandstorm Wekan, just made more previously hidden to be now visible. Code for Sandstorm features etc is still the same.

More info about features:
https://github.com/wekan/wekan/blob/devel/CHANGELOG.md#v228-2019-02-27-wekan-release

So now that when making new Wekan board, first user will be made Admin, and can see Admin Panel of Standalone Wekan. This adds are some open questions:
- What about old Wekan grains, where is no Admin user? Should I try to make everyone in old grains admin?
- Does Wekan REST API work now with Admin user somehow?
- Do those IFTTT Rules work now? Note, that Wekan grain is running only when some user has grain open. Is that enough? Wekan does not yet have scheduler/cron like meteor package sjobs, but it will be added sometime.
- When adding new users to board, can Admin user change those new user roles Admin/User/NoComments/CommentOnly with webbrowser or [REST API](https://github.com/wekan/wekan/wiki/REST-API-Role) ?
- Etc. What does not work yet?

ping @ocdtrekkie @ertanalytics @kentonv @JamborJan - if you have time and are interested

Anyone please add feedback etc, so this gets tested well.

**Comment by ocdtrekkie** [comment]
Should probably mention, it's on the experimental market, at the moment, which is here: https://apps.sandstorm.io/app/m86q05rdvj14yvn78ghaxynqz7u2svw6rnttptxx49g1785cdv1h?experimental=true

**Comment by ocdtrekkie** [comment]
As a note, for Wekan on Sandstorm to be able to use a Cron-like scheduler, Sandstorm needs to implement a Cron API (Unloved PR here: https://github.com/sandstorm-io/sandstorm/pull/2922) and then Wekan would need to implement it, so that the Sandstorm grain is awoken.

**Comment by JamborJan** [comment]
Hey guys,

Thanks for the info and the great work. As far as I can plan I'll not be able to do any tests in the upcoming 2 months. Sorry.

JJ

________________________________
Von: Jacob Weisz <notifications@github.com>
Gesendet: Mittwoch, Februar 27, 2019 6:03 PM
An: wekan/wekan
Cc: Jan Jambor; Mention
Betreff: Re: [wekan/wekan] Everyone/Anyone please test new Sandstorm Wekan features (#2208)


As a note, for Wekan on Sandstorm to be able to use a Cron-like scheduler, Sandstorm needs to implement a Cron API (Unloved PR here: sandstorm-io/sandstorm#2922<https://github.com/sandstorm-io/sandstorm/pull/2922>) and then Wekan would need to implement it, so that the Sandstorm grain is awoken.

**Comment by ocdtrekkie** [comment]
Subtasks doesn't work (creating one seems to happen then immediately disappears). I know Subtasks didn't work on Sandstorm before, so I'm not sure if it was disabled somehow.

**Comment by bsenicar** [comment]
Sending mail from IFTTT rules does not work.

====== BEGIN MAIL #0 ======
(Mail not sent; to enable sending, set the MAIL_URL environment variable.)
Content-Type: text/plain
From: Boards Support <support@127.0.0.1:4000>
Message-ID: <7f6d8205-f405-7600-bf54-01f25fd133cd@127.0.0.1:4000>
Date: Mon, 04 Mar 2019 10:58:45 +0000
MIME-Version: 1.0
====== END MAIL #0 ======

Wekan version: 2.37.0~2019-03-04 from experimental market.

BR
Borut

**Comment by xet7** [comment]
@bsenicar 

Sending mail from Sandstorm Wekan is not implemented yet, I would need to find some documentation how to send email from sandboxed Sandstorm grain.

**Comment by xet7** [comment]
Hmm, I think at Sandstorm it's not possible to change environment variables, so some settings would probably need to be added to Wekan Admin Panel.

**Comment by ocdtrekkie** [comment]
https://docs.sandstorm.io/en/latest/developing/email-from-apps/

**Comment by xet7** [comment]
Moved to here from #2388 

From @SteamWind 

Hello,
Using wekan (v2.61.0) on sandstorm, I've got an error when I try to edit the users on the *Admin Panel->People* view (when clicking on save after edit).
I seems that the Emails array is not defined on [this line](https://github.com/wekan/wekan/blob/a5122cc0764b57a163c3e8ca3e00e703669a98db/client/components/settings/peopleBody.js#L161)

Here is the (anonimised) object that is used to get Emails:

```javascript
{
    "_id":"RaFkeidTdTcvbWeveT",
    "createdAt":"2019-04-26T10:40:07.351Z",
    "profile":{
        "fullname":"Full Name"
    },
    "authenticationMethod":"password","username":"full_name"
}
```
Full error:
```
Uncaught TypeError: Cannot read property '0' of undefined
    at Object.submit (dde565b9b359dceb4eaad44288690fbdbbb476c2.js?meteor_js_resource=true:formatted:99453)
    at dde565b9b359dceb4eaad44288690fbdbbb476c2.js?meteor_js_resource=true:formatted:20218
    at Function.e._withTemplateInstanceFunc (dde565b9b359dceb4eaad44288690fbdbbb476c2.js?meteor_js_resource=true:formatted:20197)
    at f.View.<anonymous> (dde565b9b359dceb4eaad44288690fbdbbb476c2.js?meteor_js_resource=true:formatted:20217)
    at dde565b9b359dceb4eaad44288690fbdbbb476c2.js?meteor_js_resource=true:formatted:19685
    at Object.f._withCurrentView (dde565b9b359dceb4eaad44288690fbdbbb476c2.js?meteor_js_resource=true:formatted:19496)
    at f._DOMRange.<anonymous> (dde565b9b359dceb4eaad44288690fbdbbb476c2.js?meteor_js_resource=true:formatted:19684)
    at HTMLFormElement.<anonymous> (dde565b9b359dceb4eaad44288690fbdbbb476c2.js?meteor_js_resource=true:formatted:18841)
    at HTMLDivElement.dispatch (dde565b9b359dceb4eaad44288690fbdbbb476c2.js?meteor_js_resource=true:formatted:11194)
    at HTMLDivElement.g.handle (dde565b9b359dceb4eaad44288690fbdbbb476c2.js?meteor_js_resource=true:formatted:11089)
```

**Comment by kentonv** [comment]
I just encountered these new features, and I'm a bit sad that it means Wekan no longer adheres to the Sandstorm fine-grained model. Now a grain may contain multiple boards with I guess Wekan doing its own internal access control on boards. I think this is confusing to users as now there are two different sharing models.

That said, obviously the lack of any development on Sandstorm itself must be frustrating for app developers that want to provide more advanced sharing features. So I'm not sure if I have much right to complain.

Incidentally, an Oasis user complained to me today that the "add member" button does not work on secondary boards, and indeed this seems to be the case from my testing.

**Comment by xet7** [comment]
@kentonv 

Hehe, I think there is no reason to feel sad when I add new features and fixes to Sandstorm Wekan.

I will continue adding features and fixes to Wekan. If some change is needed for Sandstorm also, I will send PR to Sandstorm.

Add member issue is #2405 . I will add fix when I get it working.

**Comment by xet7** [comment]
@kentonv 

Would it be better, that at Sandstorm I would move Wekan Admin Panel features to Sandstorm Admin Panel?

**Comment by ocdtrekkie** [comment]
@kentonv It actually wasn't mostly sharing features held up by the single-board design! Subtasks and board importing and the like were all hung-up on it, since they used additional boards to make them work.

#2405 is actually held up primarily on the issue that nobody who got the Sandstorm sharing permissions working still works on Wekan. (Was mquandalle and dwrensha, I think.) The ability to share Wekan admin is listed in the pkgdef but commented out, and I imagine re-adding it would give us the solution there. Sandstorm is still controlling Wekan permissions for now... it just only applies to the initial/first board.

**Comment by kentonv** [comment]
Yeah in an ideal world, Wekan would be able to ask Sandstorm to create those additional boards and then they'd communicate over capability-based RPC, so that each board would still be a separate grain.

But in reality there's no one working on that on the Sandstorm side, so I guess compromise is needed...
